Platon/backend/app/activity_stream/views.py

```diff
@@ -0,0 +1,212 @@
+    @api.response(200, 'Success', activity_stream_model)
+    @api.expect(get_activity_stream_parser)
+    @login_required
+    def get(user_id, self):
+        '''
+            Returns activity stream of an user.
+        '''
+        '''
+            # Pagination functionality
+            number_of_pages = 1
+            if form.page.data is not None and form.per_page.data is not None:
+                per_page = form.per_page.data
+                number_of_pages = math.ceil(len(return_list) / per_page)
+                # Assign the page index to the maximum if it exceeds the max index
+                page = form.page.data if form.page.data < number_of_pages else number_of_pages-1
+                return_list = return_list[page*per_page:(page+1)*per_page]
+            return make_response(jsonify({'number_of_pages': number_of_pages ,'result': return_list}),200)
+        '''
+
+        # Parses the form data.
+        form = GetActivityStreamForm(request.args)
+        
+        if form.validate():
+
+            # Get current user's followings list.
+            try:
+                followings_list = Follow.query.filter(Follow.follower_id==user_id).all()
+            except:
+                return make_response(jsonify({"error" : "The server is not connected to the database."}), 500)
+
+            followings_ids_list = []
+            for followings_item in followings_list:
+                followings_ids_list.append(followings_item.following_id)
+
+            # Get current user's workspaces list.
+            try:
+                contributions = Contribution.query.filter(Contribution.user_id==user_id).all()
+            except:
+                return make_response(jsonify({"error": "Database Connection Error"}),500)
+            
+            workspace_ids_list = []
+            for contribution in contributions:
+                if contribution.is_active == 0:
+                    continue
+                try:
+                    ws = Workspace.query.get(contribution.workspace_id)
+                except:
+                    return make_response(jsonify({"error": "Database Connection Error"}),500)
+
+                workspace_ids_list.append(ws.id)
+
+            activity_stream_list = []
+            for following_id in followings_ids_list:
+                try:
+                    activity_stream_items_list = ActivityStreamItem.query.filter(ActivityStreamItem.activity_actor_id == following_id).all()
+                except:
+                    # not sure to return error.
+                    return make_response(jsonify({"error" : "The server is not connected to the database."}), 500)
+
+                activity_stream_list += activity_stream_items_list
+
+            # Sort all activities.
+            activity_stream_list.sort(key = lambda activity_stream_item: activity_stream_item.timestamp, reverse=True)
```

Comment:
There is no need to sort these items on the backend server. You can take them from the database in an ordered way by using order_by function of the sqlalchemy models. 

You can use [this link](https://stackoverflow.com/questions/4186062/sqlalchemy-order-by-descending) for more information.